### PR TITLE
APP-11219 web component description list

### PR DESCRIPTION
**Pull request by Sqrrl**
## Summary

- [Ticket](https://flipapp.myjetbrains.com/youtrack/issue/APP-11219/Web-Component-Description-List-formerly-Data-Display)
- [Design](https://www.figma.com/file/iRHiAFsHxnzux8gQ9EjlZe/Flip-Components?node-id=2%3A22)
- Storybook: [FlipDescriptionList](https://gentle-water-054ae5203-36.westeurope.1.azurestaticapps.net/?path=/docs/components-flipdescriptionlist--flip-description-list) + [FlipDescriptionListItem](https://gentle-water-054ae5203-36.westeurope.1.azurestaticapps.net/?path=/docs/components-flipdescriptionlistitem--flip-description-list-item)

### General

- [ ] [Changeset added](https://github.com/changesets/changesets/blob/main/docs/intro-to-using-changesets.md)
- [ ] The submitted code is organized and formatted according to our [💅 Style Guide](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-style-guide--page).
  - [ ] The code is formatted with Prettier.
  - [ ] There are no linting errors.

### For new or updated components

https://swirl-storybook.flip-app.dev/?path=/docs/contributions-merge-publish--page

- [ ] The changes do not contain any breaking changes.
- [ ] The changes do not introduce new components that don't belong in the library (e.g. non-reusable components, highly specialized components, components including business logic)
- [ ] The component documentation is updated.
- [ ] The changes meet the [🤖 testing requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-testing--page).
  - [ ] New features are tested.
  - [ ] In case of bug fixes, regression tests have been added.
  - [ ] All tests are 🟢.
- [ ] The changes meet the [🧏‍♀️ accessibility requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-accessibility--page).
  - [ ] WCAG 2.1 Level A and Level AA requirements are met.
  - [ ] The Storybook a11y addon shows no errors.
  - [ ] The changes have been tested with a screen reader.
  - [ ] Keyboard controls have been tested, if applicable.
  - [ ] Components implementing form controls (inputs, buttons, selects, etc.) do not use Shadow DOM, and instead use Stencil's scoping mechanism
- [ ] The changes use our [🌈 theming concept](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-theming--page).
  - [ ] Design tokens have been used where appropriate.
  - [ ] The component has been visually checked in combination with the "Light" and "Dark" theme.
- [ ] The changes meet our [🌍 internationalization requirements](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-internationalization--page).
  - [ ] No static text is used.
  - [ ] The component doesn't break with longer texts or different text wrappings.
  - [ ] Number, currency and date values can be formatted appropriately.
- [ ] The changes work in all supported browsers and viewports. See [📱 Responsive Design](https://swirl-storybook.flip-app.dev/?path=/docs/requirements-responsive-design--page)

### CODE REVIEW

File: packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+@container description-list (min-width: 480px) {
+  .description-list-item {
+    display: flex;
+  }
+
+  .description-list-item__term {
+    margin-right: var(--s-space-16);
+    margin-bottom: 0;
```
Comment (webspaceadam) [inline packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css:37]: just personal taste but i think it is better to have display options (flex stuff) above the box-model stuff (margin)

Comment (Sqrrl) [reply packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css:37]: That's from my current postcss-sort setup. Let's adjust this, once we have figured out a shared sort config.

Comment (webspaceadam) [reply packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css:37]: 👍

File: packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  .description-list-item__description {
+    flex-grow: 1;
+  }
+}
+
+/* Fallback styles for browsers without container query support */
+@supports not (container-type: inline-size) {
+  @media (--from-tablet) {
+    .description-list-item {
+      display: flex;
```
Comment (webspaceadam) [inline packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css:52]: when i look at the design i would argue that we should use <code>justify-content: space-between</code> here?
image

Comment (webspaceadam) [reply packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css:52]: at least this looks *"correcter"* to me ^^
image

**Review by webspaceadam — CHANGES_REQUESTED** [review]

File: packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css (hunk 2 above)
Comment (Sqrrl) [reply packages/swirl-components/src/components/flip-description-list-item/flip-description-list-item.css:52]: I would argue that this creates too much white space between the term and the description on wide screens, doesn't it?

**Review by webspaceadam — APPROVED** [review]

**Pull request merged**